### DISCUSSION TITLE
What is the use of the swift package within ./apple/MapLibreRs/

### DISCUSSION THREAD

**Discussion by Drabble** [maplibre/maplibre-rs#56] [Q&A] [answered]
I was looking at the `./apple` folder and stumbled upon the folder named `MapLibreRs`. I don't have much experience with iOS development so to me it looked like a forgotten test Swift package. After some research I found that `MapLibreRs` is used as the output for the XCFramework build in the Github actions. I also found [in the Apple documentation](https://developer.apple.com/documentation/swift_packages/distributing_binary_frameworks_as_swift_packages) that you can wrap a XCFramework build inside another Swift package. Could you tell me why this is useful or what I have not understood correctly?

In addition, it seems to be the only folder that contains upper-case characters.

Maybe I could write some documentation in the MapLibreRs README for the newcomers, if I understand how it works.

**Comment by maxammann** [comment, marked as answer]
So I'm also not at all familiar with apple. But it seems like having an xcframework is not really enough to distribute it properly. 

One way to distribute seems to be to host the xcframework as part of a swift package. Here is how maplibre-gl is doing it: https://github.com/maplibre/maplibre-gl-native-distribution

Swift packages are commonly written im uppercase CamelCase. So MaplibreRs is the consequence. Im not sure here though.


All of the above statements are very vague as I never deployed an xcode binary framework.

**Reply by maxammann** [reply]
In this task we woulf distribute the swift package: https://github.com/maplibre/maplibre-rs/issues/21

Description is still missing though
